src/v/storage/log_manager.cc

```diff
@@ -127,31 +130,49 @@ ss::future<> log_manager::housekeeping() {
      * them as fast as we can since the majority of the time they will be in the
      * hotpath / (request-response)
      */
+    std::unique_ptr<model::ntp> current_key{};
     using bflags = log_housekeeping_meta::bitflags;
-    return ss::do_until(
```

Comment:
since nearly this entire function is being refactored, it might make sense to switch over to coroutines.

Reply:
Went ahead and refactored it into a coroutine. Thanks!